src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1482,6 +1484,7 @@ public void addVisitingPlayer(final PlayerEntity player)
     @Override
     public void removeVisitingPlayer(final PlayerEntity player)
     {
+        ItemBlockHut.checkResearch(null);
```

Comment:
this will be done on each chunk a player enters, might be a bit too much?

Reply:
I think this will only be done once, when the player leaves the colony.

Reply:
We do the list checks a lot here. So I'm not sure if this is called on every entry and exit of a chunk.

Reply:
I have looked at every method calling this, and to me it looks like this is only called if an entity leaves a chunk into one that is claimed by another colony(or none), and if a player leaves a dimension.

Reply:
Then it's fine.